soh/soh/OTRGlobals.cpp

```diff
@@ -354,7 +354,7 @@ OTRGlobals::OTRGlobals() {
     overlay->LoadFont("Fipps", 32.0f, "fonts/Fipps-Regular.otf");
     overlay->SetCurrentFont(CVarGetString(CVAR_GAME_OVERLAY_FONT, "Press Start 2P"));
 
-    context->InitAudio({ .SampleRate = 44100, .SampleLength = 1024, .DesiredBuffered = 2480 });
+    context->InitAudio({ .SampleRate = 32000, .SampleLength = 1024, .DesiredBuffered = 2480 });
```

Comment:
looks like `DesiredBuffered` was changed for 44.1k [here](https://github.com/HarbourMasters/Shipwright/pull/477/files#diff-b5228ea83b193b71ef99d9212d402fdd9d273bf18c55e696547f261c1b4edf06R31-R33)

```diff
		int SDLAudioPlayer::GetDesiredBuffered(void) {
-            return 1680;
+            return 2480;
		}
```

interestingly, it seems `SampleLength` was already 1024 before the switch to 44.1k

i also am not sure where these values are coming from:
32000/1680 = 19.047619048
44100/2480 = 17.782258065

@Kenix3 @NEstelami any ideas?

Reply:
I can't find that. Where is it?

Reply:
https://github.com/HarbourMasters/Shipwright/blame/aeaaa54e0daa7482bdcdd0f26466b738b25fe4e3/libultraship/libultraship/SDLAudioPlayer.cpp#L31-L33

vs

https://github.com/HarbourMasters/Shipwright/blob/1f1b81ab4022f98e46a05b84b17de63aca0a5a15/libultraship/libultraship/SDLAudioPlayer.cpp#L31-L33

and


https://github.com/HarbourMasters/Shipwright/blame/aeaaa54e0daa7482bdcdd0f26466b738b25fe4e3/libultraship/libultraship/WasapiAudioPlayer.cpp#L83-L85

vs

https://github.com/HarbourMasters/Shipwright/blob/1f1b81ab4022f98e46a05b84b17de63aca0a5a15/libultraship/libultraship/WasapiAudioPlayer.cpp#L83-L85

Reply:
for sample length, it's used [here](https://github.com/Kenix3/libultraship/blob/2942915fc3727549806db020c9edc81aad0029d3/src/audio/SDLAudioPlayer.cpp#L21) in LUS (but i don't see it being used in [wsapi](https://github.com/Kenix3/libultraship/blob/main/src/audio/WasapiAudioPlayer.cpp))

but it was hardcoded to 1024 at the beginning of the repo being public

https://github.com/HarbourMasters/Shipwright/blob/d24c8453db1035f382e1b0853be00ebd281bbbdd/libultraship/libultraship/SDLAudioPlayer.cpp#L15

Reply:
ok, did some digging and figured out stuff about the value for samples

it looks like in SDL3 `samples` has been removed from `AudioSpec`
https://wiki.libsdl.org/SDL3/README/migration#sdl_audioh
https://wiki.libsdl.org/SDL2/SDL_AudioSpec
https://wiki.libsdl.org/SDL3/SDL_AudioSpec

from the migration guide:

> SDL_AudioSpec has been reduced; now it only holds format, channel, and sample rate. SDL_GetSilenceValueForFormat() can provide the information from the SDL_AudioSpec's removed silence field. SDL3 now manages the removed samples field; apps that want more control over device latency and throughput can force a newly-opened device's sample count with the SDL_HINT_AUDIO_DEVICE_SAMPLE_FRAMES hint, but most apps should not risk messing with the defaults. The other SDL2 SDL_AudioSpec fields aren't relevant anymore.

sdl3 logic for getting the value is here https://github.com/libsdl-org/SDL/blob/7c9f6c631304829cb5d6fc68d873dace47c17152/src/audio/SDL_audio.c#L147-L166
```c
    if (freq <= 22050) {
        return 512;
    } else if (freq <= 48000) {
        return 1024;
    } else if (freq <= 96000) {
        return 2048;
    } else {
        return 4096;
    }
```
so based on what SDL is doing it seems `1024` is a reasonable value for both 32k and 44.1k

Reply:
```suggestion
    context->InitAudio({ .SampleRate = 32000, .SampleLength = 1024, .DesiredBuffered = 1680 });
```

see

https://github.com/HarbourMasters/Shipwright/pull/4780/files?diff=unified&w=0#r1900436401
https://github.com/HarbourMasters/Shipwright/pull/4780/files?diff=unified&w=0#r1900475059